lib/parsers/m3u_parser.rb

```diff
@@ -0,0 +1,25 @@
+class FormatParser::M3UParser
+  include FormatParser::IOUtils
+
+  HEADER = '#EXTM3U'
+
+  def likely_match?(filename)
+    filename =~ /\.(m3u|m3u8)$/i
+  end
+
+  def call(io)
+    io = FormatParser::IOConstraint.new(io)
+
+    header = safe_read(io, 7)
+    return unless HEADER.eql?(header)
+
+    file_content = io.read(io.size)
+
+    FormatParser::M3U.new(
+      format: :m3u,
+      size: io.size,
+      content: file_content
+    )
+  end
+  FormatParser.register_parser new, natures: :text, formats: [:m3u, :m3u8]
```

Comment:
I'm almost sure that the `natures` must be related to a class which includes `FormatParser::AttributesJSON`, in this case, the `M3u` that you created.

It would be nice to understand if something breaks when using an "invalid" nature and if so, creating a test to cover this. (I think that "text" is an invalid nature)

Reply:
Yes, that's what I meant above with renaming `M3U` to `Text` 🙂